src/arch/src/aarch64/regs.rs

```diff
@@ -139,9 +144,51 @@ macro_rules! arm64_sys_reg {
     };
 }
 
+// This macro extracts the MANUFACTURER_ID from the provided MIDR_EL1 register value,
+// found between bits 24 and 31.
+// https://developer.arm.com/documentation/100095/0001/system-control/
+// aarch64-register-summary/aarch64-identification-registers?lang=en
+macro_rules! manufacturer_id {
+    ($midr_el1: expr) => {
+        $midr_el1 as u32 >> 24
+    };
+}
+
 // Constant imported from the Linux kernel:
 // https://elixir.bootlin.com/linux/v4.20.17/source/arch/arm64/include/asm/sysreg.h#L135
 arm64_sys_reg!(MPIDR_EL1, 3, 0, 0, 0, 5);
+arm64_sys_reg!(MIDR_EL1, 3, 0, 0, 0, 0);
+
+/// Extract the manufacturer_id register from a VCPU state's registers
+///
+/// # Arguments
+///
+/// * `state` - Array slice of kvm_one_reg structures, representing
+///             the registers of a VCPU state.
+pub fn get_manufacturer_id_from_state(state: &[kvm_one_reg]) -> Result<u32> {
+    let midr_el1 = state.iter().find(|reg| reg.id == MIDR_EL1);
+    match midr_el1 {
+        Some(value) => Ok(manufacturer_id!(value.addr as u32)),
+        None => Err(Error::GetMidrEl1(
+            "Failed to get MIDR_EL1 from VCPu state!".to_string(),
+        )),
+    }
+}
+
+/// Extract the Manufacturer ID from the host
+pub fn get_manufacturer_id_from_host() -> Result<u32> {
+    let midr_el1_path = &PathBuf::from(format!(
+        "/sys/devices/system/cpu/cpu0/regs/identification/midr_el1"
+    ));
+
+    let midr_el1 = fs::read_to_string(midr_el1_path)
```

Comment:
If this fails the error will look something like this:
"Load microVM snapshot error: Invalid aarch64 register: Failed to get MIDR_EL1: Failed to get MIDR_EL1 from host path!"
It does not tell anywhere that it failed while trying to match the manufacturer ids.
Now my proposal would be:
1. Make use of the already existent error: "CpuVendorMismatch" for "validate_aarch64_cpu_manufacturer_id" too:
```
let host_man_id =
        get_manufacturer_id_from_host().map_err(|e|LoadSnapshotError::CpuVendorMismatch(e.to_string()))?;
```
2. the display trait for  GetMidrEl1 is changed to:
 ```
self::Error::GetMidrEl1(ref e) => write!(f, "{}", e),
```
3. This line becomes: 
```
let midr_el1 = fs::read_to_string(midr_el1_path)
        .map_err(|e| Error::GetMidrEl1(format!("Failed to get MIDR_EL1 from host path: {}", e.to_string())))?;
```
The error thrown if read_to_string fails would be this:
"Load microVM snapshot error: Snapshot cpu vendor mismatch: Failed to get MIDR_EL1 from host path: No such file or directory (os error 2)".
WDYT?

Reply:
Sounds good! Will do as such.